src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2163,6 +2162,52 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                 return;
             }
 
+            if (reason == LambdaConversionResult.MismatchedParameterDefaultValue)
+            {
+                Debug.Assert(anonymousFunction.ParameterCount == delegateParameters.Length);
+
+                var lambdaSymbol = anonymousFunction.TemporaryLambdaSymbol;
+                Debug.Assert(lambdaSymbol is not null);
+
+                // The lambda symbol may have diagnostics from
+                // binding default parameters, so copy these to the current diagnostic bag
+                lambdaSymbol.GetDeclarationDiagnostics(diagnostics);
+
+                for (int i = 0; i < anonymousFunction.ParameterCount; i++)
+                {
+
+                    var lambdaParameterLocation = anonymousFunction.ParameterLocation(i);
+                    var lambdaParamDefaultVal = lambdaSymbol.Parameters[i].ExplicitDefaultConstantValue;
+                    var delegateParamDefaultVal = delegateParameters[i].ExplicitDefaultConstantValue;
+
+                    if (lambdaParamDefaultVal is not null && delegateParamDefaultVal is not null && lambdaParamDefaultVal != delegateParamDefaultVal)
```

Comment:
We can simplify this to just `lambdaDefaultVal != delegateParamDefaultVal` if we remove the extra conversion result.

Reply:
We still need to check if `lambdaParamDefaultVal is not null`, because if it is null then we are in an already-supported scenario and we don't want to throw the error

Reply:
I wouldn't expect to get into this codepath at all if it's that case, as the lambda binding will have been a success.